### PR TITLE
More review fixes: remove cache wrapper, better view creation, applied trivial fixes, applied required renamings

### PR DESCRIPTION
**Pull request by ariostas**
- Removed `allocBufWrapper` and cache toggle.
- Fixed failing header test.
- Improved view creation.
- Removed `devAcc_`.
- Applied trivial fixes.
- Applied required renamings.

### CODE REVIEW

File: RecoTracker/LSTCore/src/alpaka/Event.dev.cc [hunk 1]
```diff
@@ -696,18 +696,18 @@ void Event::createTrackCandidates(bool no_pls_dupclean, bool tc_pls_triplets) {
                       tc_pls_triplets);
 
   // Check if either n_max_pixel_track_candidates or n_max_nonpixel_track_candidates was reached
-  auto nTrackCanpT5Host_buf = allocBufWrapper<unsigned int>(cms::alpakatools::host(), 1, queue_);
-  auto nTrackCanpT3Host_buf = allocBufWrapper<unsigned int>(cms::alpakatools::host(), 1, queue_);
-  auto nTrackCanpLSHost_buf = allocBufWrapper<unsigned int>(cms::alpakatools::host(), 1, queue_);
-  auto nTrackCanT5Host_buf = allocBufWrapper<unsigned int>(cms::alpakatools::host(), 1, queue_);
+  auto nTrackCanpT5Host_buf = cms::alpakatools::make_host_buffer<unsigned int[]>(queue_, 1u);
```
Comment (ariostas) [inline RecoTracker/LSTCore/src/alpaka/Event.dev.cc:699]: For scalars, we could use
```
cms::alpakatools::make_host_buffer<unsigned int>(queue_);
```

The issue is that I couldn't find an easy way to create a scalar view for the scalars in SoAs, so `memcpy` fails to compile since they have different dimensions.

Comment (ariostas) [reply RecoTracker/LSTCore/src/alpaka/Event.dev.cc:699]: Actually, I figured out how to do this

File: RecoTracker/LSTCore/interface/EndcapGeometryDevSoA.h [hunk 2]
```diff
@@ -4,6 +4,8 @@
 #include "DataFormats/SoATemplate/interface/SoALayout.h"
 #include "DataFormats/Portable/interface/PortableCollection.h"
 
+#include "RecoTracker/LSTCore/interface/Common.h"
```
Comment (VourMa) [inline RecoTracker/LSTCore/interface/EndcapGeometryDevSoA.h:7]: Why is this needed? Just for clarity?

Same for the rest of the files.

File: RecoTracker/LSTCore/interface/LSTESData.h [hunk 3]
```diff
@@ -21,7 +21,8 @@ namespace lst {
     unsigned int nEndCapMap;
     std::unique_ptr<const PortableMultiCollection<TDev, ModulesSoA, ModulesPixelSoA>> modules;
     std::unique_ptr<const PortableCollection<EndcapGeometryDevSoA, TDev>> endcapGeometry;
-    std::shared_ptr<const PixelMap> pixelMapping;
+    std::shared_ptr<const PixelMap>
+        pixelMapping;  // Host-side object that is shared between the LSTESData<TDev> objects for different devices
```
Comment (VourMa) [inline RecoTracker/LSTCore/interface/LSTESData.h:25]: Can you put this comment on a separate line just above to avoid the awkward line break?

**Review by VourMa — COMMENTED** [review]
Thanks, Andres, I only have very minor comments (mostly for my understanding). Let's see how the tests run and I can merge this rather quickly.

In the meantime, could you update the PR description to be a bit more descriptive (don't go as far as I did in the Google document) and add a comment in the Google document about which comment you addressed?

File: RecoTracker/LSTCore/interface/Common.h [hunk 4]
```diff
@@ -12,35 +12,8 @@
 #endif
 #endif
 
-#ifdef CACHE_ALLOC
-#include "HeterogeneousCore/AlpakaInterface/interface/CachedBufAlloc.h"
```
[the remark this answers is not in the document]
Comment (slava77) [reply RecoTracker/LSTCore/interface/Common.h:16]: this comes via HeterogeneousCore/AlpakaInterface/interface/memory.h
and also DataFormats/Portable includes.
So, should be OK. Although pay attention to the header check in cms bot tests.

File: RecoTracker/LSTCore/interface/EndcapGeometryDevSoA.h (hunk 2 above)
Comment (ariostas) [reply RecoTracker/LSTCore/interface/EndcapGeometryDevSoA.h:7]: In this particular case it's not needed, but in some of them we need it to use array types defined in that header. I ended up just adding it to all of them, but I can remove the unneeded ones

File: RecoTracker/LSTCore/interface/LSTESData.h (hunk 3 above)
Comment (ariostas) [reply RecoTracker/LSTCore/interface/LSTESData.h:25]: Yeah, sounds good

File: RecoTracker/LSTCore/interface/Common.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (ariostas) [reply RecoTracker/LSTCore/interface/Common.h:16]: Yeah, I fixed the failing header tests. We should also check that in our CI.

File: RecoTracker/LSTCore/interface/EndcapGeometryDevSoA.h (hunk 2 above)
Comment (VourMa) [reply RecoTracker/LSTCore/interface/EndcapGeometryDevSoA.h:7]: No need, let's leave it like this. Thanks!

**Comment by ariostas** [comment]
Thank you @VourMa! I addressed your comments. Let me know if there's anything else

File: RecoTracker/LSTCore/interface/EndcapGeometryDevSoA.h (hunk 2 above)
Comment (ariostas) [reply RecoTracker/LSTCore/interface/EndcapGeometryDevSoA.h]: Also, for the DeviceCollection ones it's needed so that it knows that `ALPAKA_ACCELERATOR_NAMESPACE::lst` reuses everything in `lst`.

**Comment by slava77** [comment]
I scrolled through the updates as well; looks good to me.

**Review by VourMa — APPROVED** [review]
Thank you @ariostas, I am merging this!
